packages/main/src/Menu.js

```diff
@@ -0,0 +1,479 @@
+	}
+
+	static get dependencies() {
+		return [
+			ResponsivePopover,
+			Button,
+			List,
+			StandardListItem,
+			Icon,
+		];
+	}
+
+	static async onDefine() {
+		Menu.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	get itemsWithChildren() {
+		return !!this._currentItems.filter(item => item.items.length).length;
+	}
+
+	get itemsWithIcon() {
+		return !!this._currentItems.filter(item => item.icon !== "").length;
+	}
+
+	get isRtl() {
+		return this.effectiveDir === "rtl";
+	}
+
+	get placementType() {
+		const placement = this.isRtl ? "Left" : "Right";
+		return this._isSubMenu ? placement : "Bottom";
+	}
+
+	get verticalAlign() {
+		return this._isSubMenu ? "Top" : "Bottom";
+	}
+
+	get labelBack() {
+		return Menu.i18nBundle.getText(MENU_BACK_BUTTON_ARIA_LABEL);
+	}
+
+	get labelClose() {
+		return Menu.i18nBundle.getText(MENU_CLOSE_BUTTON_ARIA_LABEL);
+	}
+
+	get isPhone() {
+		return isPhone();
+	}
+
+	get isSubMenuOpened() {
+		return !!this._parentMenuItem;
+	}
+
+	get menuTitlePhone() {
+		return this.isSubMenuOpened ? this._parentMenuItem.text : this.title;
+	}
+
+	onBeforeRendering() {
+		!isPhone() && this._prepareCurrentItems();
+
+		const itemsWithChildren = this.itemsWithChildren;
+		const itemsWithIcon = this.itemsWithIcon;
+
+		this._currentItems.forEach(item => {
+			item._siblingsWithChildren = itemsWithChildren;
+			item._siblingsWithIcon = itemsWithIcon;
+		});
+	}
+
+	/**
+	 * Shows the Menu near the opener element.
+	 * @param {HTMLElement} opener the element that the popover is shown at
+	 * @public
+	 */
+	async showAt(opener) {
+		if (isPhone()) {
+			this._prepareCurrentItems();
+			this._parentItemsStack = [];
+		}
+		if (!this._isSubMenu) {
+			this._parentMenuItem = undefined;
+		}
+		(await this._getPopover()).showAt(opener);
+	}
+
+	/**
+	 * Closes the Menu.
+	 * @param {HTMLElement} opener the element that the popover is shown at
+	 * @public
+	 */
+	close() {
+		if (isPhone()) {
+			this._parentItemsStack = [];
+		}
+		this._popover.close();
+		this._popover.resetFocus();
+	}
+
+	async _getPopover() {
+		const id = this._id;
+		this._popover = (await this.getStaticAreaItemDomRef()).querySelector("[ui5-responsive-popover]");
+		this._popover.initialFocus = `${id}-menu-item-0`;
+		return this._popover;
+	}
+
+	_beforePopoverClose() {
+		if (Object.keys(this._openedSubMenuItem).length) {
+			this._openedSubMenuItem._preventSubMenuClose = false;
+			this._closeItemSubMenu(this._openedSubMenuItem);
+		}
+	}
+
+	_navigateBack(event) {
+		const parentMenuItem = this._parentItemsStack.pop();
+		this.focus();
+		if (parentMenuItem) {
+			this._currentItems = parentMenuItem.parentElement.items;
+			this._parentMenuItem = this._parentItemsStack.length ? this._parentItemsStack[this._parentItemsStack.length - 1] : undefined;
+		}
+	}
+
+	_prepareCurrentItems() {
+		this._currentItems = this.items;
+	}
+
+	_createSubMenu(item, openerId) {
+		const subMenu = document.createElement("ui5-menu");
```

Comment:
This whole method is concerning - at first glance it seems that what you need is a static area template, where this sub menu addition is going to happen. You cannot append to the body as it belongs to the app. For all such purposes you must use the static area. Also, you cannot use `document.createElement`, you must use the .hbs file for creating DOM. Finally, hard-coded `ui5-menu` is not scoping-friendly - if the user set a scoping suffix such as `test`, it would be f.e. `ui5-menu-test`. When you use the .hbs, the compiler will do this automatically. However, this component creates other instances of itself, so I'm not even sure this is going to work.

Reply:
I'm not sure it can be done via hbs template. In fact, a new ui5-menu element must be created for each nested submenu.